test/jdk/java/awt/TrayIcon/DisposeInActionEventTest/DisposeInActionEventTest.java

```diff
@@ -21,76 +21,77 @@
  * questions.
  */
 
-
-/*
-  test
-  @bug 6299866
-  @summary Tests that no NPE is thrown when the tray icon is disposed from the
-  handler of action event caused by clicking on this icon.
-  @library ../../regtesthelpers
-  @build Sysout
-  @author artem.ananiev: area=awt.tray
-  @run applet/manual=yesno DisposeInActionEventTest.html
-*/
-
-import java.applet.*;
-
-import java.awt.*;
-import java.awt.image.*;
+import java.awt.AWTException;
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Graphics;
+import java.awt.SystemTray;
+import java.awt.TrayIcon;
+import java.awt.image.BufferedImage;
+import javax.swing.JFrame;
 
 import jdk.test.lib.Platform;
 import test.java.awt.regtesthelpers.Sysout;
 
-public class DisposeInActionEventTest extends Applet {
-    private boolean traySupported;
-
-    private SystemTray systemTray;
-    private TrayIcon trayIcon;
-
-    public void init() {
-        this.setLayout(new BorderLayout());
+/*
+ * @test
+ * @bug 6299866
+ * @summary Tests that no NPE is thrown when the tray icon is disposed from the
+ * handler of action event caused by clicking on this icon.
+ * @library ../../regtesthelpers /test/lib
+ * @build Sysout jtreg.SkippedException
+ * @run main/manual DisposeInActionEventTest
+ */
+public class DisposeInActionEventTest {
+    private static boolean traySupported;
 
-        String[] instructions;
+    public static void main(String[] args) throws Exception {
+        String instructions;
         traySupported = SystemTray.isSupported();
         if (!traySupported) {
```

Comment:
`SystemTray.isSupported()` can be directly evaluated in if condition leading us to remove `traySupported` variable.